packages/waveform-progress/src/waveform-progress.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  LitElement,
+  html,
+  css,
+  customElement,
+  property,
+  PropertyValues,
+  TemplateResult,
+} from 'lit-element';
+import ZoneOfSilence from './models/zone-of-silence';
+
+@customElement('waveform-progress')
+export default class WaveformProgress extends LitElement {
+  @property({ type: Number }) percentComplete = 0;
+  @property({ type: String }) waveformUrl = '';
+  @property({ type: Boolean }) interactive = false;
+  @property({ type: Array }) zonesOfSilence = [];
+
+  // This is our internal, canonical source for the `percentComplete`.
+  // The public `percentComplete` will be getting modified by outside modifiers
+  // like the audio player, but since the user can scrub through the waveform,
+  // we need to be able to control when that value gets updated
+  @property({ type: Number }) private _percentComplete = 0;
+
+  private _userIsInteracting = false;
+
+  render() {
+    return html`
+      <div class="container">
+        <div id="fill" style="width: ${this._percentComplete}%"></div>
```

Comment:
QUESTION: Are there any problems with using an `id` here? Is it namespaced to just a single instance of the web component?

Reply:
Yeah everything inside here is namespaced to the web component's shadow root so it won't matter if there are multiples of these in the DOM.